WordPress/Classes/ViewRelated/Reader/Subscriptions/ReaderSubscriptionHelper.swift

```diff
@@ -30,14 +37,19 @@ struct ReaderSubscriptionHelper {
         })
     }
 
+    @MainActor
     func unfollow(_ site: ReaderSiteTopic) {
+        let generator = UINotificationFeedbackGenerator()
+        generator.prepare()
+
         NotificationCenter.default.post(name: .ReaderTopicUnfollowed, object: nil, userInfo: [ReaderNotificationKeys.topic: site])
         let service = ReaderTopicService(coreDataStack: contextManager)
         service.toggleFollowing(forSite: site, success: { _ in
-            // Do nothing
+            generator.notificationOccurred(.success)
         }, failure: { _, error in
             DDLogError("Could not unfollow site: \(String(describing: error))")
             Notice(title: ReaderFollowedSitesViewController.Strings.failedToUnfollow, message: error?.localizedDescription, feedbackType: .error).post()
+            generator.notificationOccurred(.error)
         })
     }
```

Comment:
I'm not so sure about these haptic feedback changes. If you rapidly swipe-to-unfollow some tags, will that also result in multiple haptic feedback?

In theory, you can call these `unfollow` functions in a loop, will that result in multiple haptic feedback?

I'm in a library and didn't bring my test device with me, so I couldn't test it in a real device atm...

Reply:
I'm off for today, but that's a good point. I'm not sure "success" is often used for destructive operations, so it might be worth removnig it. But there needs to be an error feedback, but it needs to coalsesce.

Reply:
Turns out, iPads doesn't have haptics – I never paid attention to it before.
But these helper methods will also need to work for iPhone in the future project, so I updated:

- I removed `.success` feedbacks from the destructive actions, which is consistent with how Apple does it. It would feel odd if there was one.
- I kept the `.error` feedback for failures, but removed the code `notificationOccurred(.error)` from code that uses `Notice(:)`. I didn't notice, but apparently they already perform feedback.